src/board.rs

```diff
@@ -11,22 +11,25 @@ impl Board {
         Self { squares }
     }
 
-    pub fn get_piece_at(&self, position: (usize, usize)) -> Option<&dyn Piece> {
-        let (x, y) = position;
-        self.squares[x][y].as_deref()
+    pub fn get_piece(&self, position: Position) -> Result<Option<&dyn Piece>, ChessError> {
+        if position.x > 7 || position.y > 7 {
```

Comment:
Probably also want to error if it is < 0

Reply:
x and y are usizes, so this would never happen